### ISSUE TITLE
Remove InteractiveReporter Project Roles from users

### ISSUE THREAD

**Issue by lucyb** [opensafely-core/job-server#4469]
OpenSAFELY Interactive isn't currently being used. We've recently removed the global `InteractiveReporter` role from anyone outside of the team, so we should do the same for project level roles too.

However, before doing this we should check whether removing the role removes access to being able to see any existing Interactive reports as we do still want people to be able do this. I believe that anyone with the `release_file_view` permission should be able to see them, which includes anyone with the `ProjectCollaborator` role, but that should be checked first.

An alternative approach might be to remove the `analysis_request_create` and `analysis_request_view` permissions from the `InteractiveReporter` role instead.

We also need to understand who has this role and whether any external users should be informed before it's removed from them. I don't think this will be the case, as no one will be expecting to use Interactive right now, but we should check before we take action.

**Comment by Jongmassey** [comment]
I have tried removing the `InteractiveReporter` role from my user and can still view [this](https://jobs.opensafely.org/opensafely-internal/interactive/testing-analysis-01H2X0PX5V5P03PHK8WGSACAYD/) interactive report, which I found via the [issues list here](https://github.com/ebmdatalab/publications-copiloted/issues?q=is%3Aissue+is%3Aclosed+interactive).

I also tried [this report](https://jobs.opensafely.org/the-impact-of-covid-19-on-prescribing-of-antimicrobials/interactive/doxycycline-tetracyclines-chronic-obstructive-pulmonary-disease-copd-codes-01HW38MAP18PWK35KP7T46DST1/) which is a project I am not a member of and that seems to b e viewable to me.

However, in both cases, regardless of my InteractiveReporter role or not, the image in these reports are missing a source e.g.

```html
<img title="Image generated from file: output/01HW38MAP18PWK35KP7T46DST1/measure_total_rate.csv">
```

Additionally, regardless of my role status - the "Generating PDF" button on the "Download report" modal is stuck on disabled (greyed out, no entry sign mouse cursor) - not sure again if this is a known issue.

Another side note, I notice my changes to my roles didn't get put in the [audit log](https://jobs.opensafely.org/staff/users/Jongmassey/audit-log/) but IIRC there's already a ticket about that somewhere.

**Comment by Jongmassey** [comment]
Thanks to a code walkthrough with @lucyb I can see how the ability to view an Interactive report comes from the [`release_file_view`](https://github.com/opensafely-core/job-server/blob/f2bae96d4cfaa009bbc92f74fa2f5612ed3a854c/jobserver/authorization/permissions.py#L11) permission, which is held by the `InteractiveReporter` and `ProjectCollaborator` [roles](https://github.com/opensafely-core/job-server/blob/f2bae96d4cfaa009bbc92f74fa2f5612ed3a854c/jobserver/authorization/roles.py). These roles can be assigned to a user both globally and on a per-project basis.

As a global `InteractiveReporter` and `ProjectCollaborator` I have tested that I can view any interactive report. When I remove both of these roles I cannot view any interactive reports. When I remove just one, I can view them again. When I add myself to a project as a `ProjectCollaborator` then I can view reports for that project and that project only. I believe this to be the desired behaviour.

I see that @lucyb  has removed the global `InteractiveReporter` role from all but a few members of Team REX. 

If we were to remove the per-project `InteractiveReporter` role from everyone then these people would be affected (by virtue of not also having the `ProjectCollaborator` role:
* lucyb : opensafely-internal

In fact, the only project `jobserver_projectmembership` rows with `InteractiveReporter` roles are lucy, Catherine, and benbc for the opensafely internal project.

Lots (n=24) of people have the global `ProjectCollaborator` role - all are bennett internal apart from orlamac who hasn't logged in to job server since Jan 2023.

The only role that has the `analysis_request_create` permission is `InteractiveReporter`. If we wanted to robustly prevent future interactive analyses being run then we could remove this permission from this role. As it stands, any `CoreDeveloper` could re-add the role globally to any user and it wouldn't appear in the audit log

`analysis_request_view` is also only held by `InteractiveReporter`. It's a bit of a misnomer as it's required to edit the metadata of a report (title and description). If we wanted users to retain this capability then we'd need to retain this role.

**Comment by lucyb** [comment]
>these people would be affected (by virtue of not also having the ProjectCollaborator role:
>     lucyb : opensafely-internal

Oh dear! :sweat_smile: 

Thanks for checking that. I guess the IG team must have removed that role from people as they closed projects. I hadn't realised.

I still think we should remove the `analysis_request_create` permission from the `InteractiveReporter` role, but write a code comment to document that we've done that. From your other investigations, it would be best that anyone trying to use Interactive spoke to Team-REX first.

**Comment by Jongmassey** [comment]
>I still think we should remove the analysis_request_create permission from the InteractiveReporter role, but write a code comment to document that we've done that

I agree

**Comment by Jongmassey** [comment]
closed via #4504
